src/core/notmuch_data_source.py

```
@@ -14,22 +14,31 @@
     import notmuch
+
     NOTMUCH_AVAILABLE = True
 except ImportError:
     notmuch = None
     NOTMUCH_AVAILABLE = False
 
 from .data_source import DataSource
+
 # Import DatabaseManager locally to avoid circular imports
+# Use TYPE_CHECKING to avoid runtime import issues
+from __future__ import annotations
```

Flake8 (7.3.0)

[error] from __future__ imports must occur at the beginning of the file

(F404)